config/routes.rb

```diff
@@ -1,11 +1,13 @@
 Rails.application.routes.draw do
 
+
   devise_for :users
   resources :users, only: [:update]
    
    resources :topics do
-     resources :posts, except: [:index]
-   end
+   resources :posts, except: [:index]
+   resources :comments, only: [:create]
```

Comment:
See the generated routes with `rake routes` and see if you have something like 'topic_post_comments'.